src/app/employee-list/employee-list.component.ts

```diff
@@ -0,0 +1,33 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { Employee } from '../model/employee';
+
+@Component({
+  selector: 'app-employee-list',
+  templateUrl: './employee-list.component.html',
+  styleUrls: ['./employee-list.component.scss']
+})
+export class EmployeeListComponent implements OnInit {
+  @Input() employeeList!: Employee[];
+  @Output() editEmployee: EventEmitter<Employee> = new EventEmitter<Employee>();
+  @Output() deleteEmployee: EventEmitter<number> = new EventEmitter<number>();
+  @Output() addEmployee: EventEmitter<string> = new EventEmitter<string>();
+  
+  displayedColumns: string[] = ['id', 'name', 'department', 'delete'];
+  
+  constructor() { }
+
+  ngOnInit(): void {
+  }
+
+  updateEmployee(employee: Employee) {
+    this.editEmployee.emit({...employee});
+  }
+
+  removeEmployee(index: number) {
+    this.deleteEmployee.emit(index);
```

Comment:
You can call the event emitters directly from the template to simplify your component. `(click)="deleteEmployee.emit(i)"`  Ideally your child components have as little logic as possible in them, with just inputs and outputs.